x/auth/ante/sigverify.go

```diff
@@ -94,6 +96,29 @@ func (spkd SetPubKeyDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
 		spkd.ak.SetAccount(ctx, acc)
 	}
 
+	// Also emit the following events, so that txs can be indexed by these
+	// indices:
+	// - signature (via `tx.signature='<sig_as_base64>'`),
+	// - address and sequence (via `message.sender=<addr> AND tx.sequence='<seq>' `).
+	sigs, err := sigTx.GetSignaturesV2()
+	if err != nil {
+		return ctx, err
+	}
+
+	var events sdk.Events
+	for _, sig := range sigs {
+		sigBz, err := signatureDataToBz(sig.Data)
+		if err != nil {
+			return ctx, err
+		}
+
+		events = append(events, sdk.NewEvent(sdk.EventTypeTx,
+			sdk.NewAttribute(sdk.AttributeKeySequence, strconv.FormatUint(sig.Sequence, 10)),
```

Comment:
I think we want to index concat(address + sequence) not just sequence. Doing a join between a `tx.sequence` + `message.sender` might be rather complex. Or is it okay?

Note that whereas we should emit a signature event for each individual single signer in a multisig, the sequence event only corresponds to the top-level multisig. Does that make sense?

Reply:
> I think we want to index concat(address + sequence) not just sequence. Doing a join between a tx.sequence + message.sender might be rather complex. Or is it okay?

By complex you mean computationally heavy? I'm not aware of how the performance of that looks like, but we do do those joins quite often, e.g. `message.sender=... AND message.action=...`.

if we do concat(address + sequence), to clients it looks like `tx.acc_seq='cosmos1abc...def24'`, do we want also a separator like `-` to make it more user-friendly?

Would also like to hear @alexanderbez's opinion on both points.

Reply:
I switched to index by addr++seq in f6e61e12b8cbc063224a36c3e313a34265f26648. I'm happy to keep either, but from a UX perspective I personally have a preference for separate tx.sequence + message.sender

Reply:
Doing a join via `AND` is not really that computationally heavy. I don't have a strong preference in which route with go with, but if we do go with concatenation (as we are now), I would highly recommend a delimiter, e.g. `- ; : /`.

Reply:
Fine with me either way

Reply:
There's no value in a sequence index on its own without address. Also I don't think message.sender will be emitted for fee.payer say so maybe it's not as reliable? If we do join let's use a separator

Reply:
ACK. @amaurym let's keep the concatenation but let's use a delimiter.

Reply:
Sounds good, I picked `/` as delimiter.